src/rulesets/Base/fastmath_able.jl

```diff
@@ -5,8 +5,33 @@ let
     # e.g. do not add `foo` unless `Base.FastMath.foo_fast` exists.
     fastable_ast = quote
         #  Trig-Basics
-        @scalar_rule cos(x) -(sin(x))
-        @scalar_rule sin(x) cos(x)
+        ## use `sincos` to comput sin and cos at the same time
+        ## for the rules for sin and cos
+        ## See issue: https://github.com/JuliaDiff/ChainRules.jl/issues/291s
+        ## sin
+        function rrule(::typeof(sin), x::Number)
+            sinx, cosx = sincos(x)
+            sin_pullback(Δy) = (NO_FIELDS, cosx * Δy)
+            return (sinx, sin_pullback)
+        end
+
+        function frule((_, Δx), ::typeof(sin), x::Number)
+            sinx, cosx = sincos(x)
+            return (sinx, cosx * Δx)
+        end
+
+        ## cos
+        function rrule(::typeof(cos), x::Number)
+            sinx, cosx = sincos(x)
+            cos_pullback(Δy) = (NO_FIELDS, -sinx * Δy)
```

Comment:
To be correct for complex numbers need to conjugate:
```suggestion
            cos_pullback(Δy) = (NO_FIELDS, -sinx' * Δy)
```